book/webpack-encore.rst

```diff
@@ -0,0 +1,148 @@
+Using Webpack Encore for your website assets
+============================================
+
+Webpack Encore is a tool to manage your javascript and css assets. It
+seamlessly integrates webpack into your symfony application.
+
+Installation
+------------
+
+First add the `WebpackEncoreBundle`_ as a dependency to your
+composer.json file.
+
+.. code:: bash
+
+   composer require symfony/webpack-encore-bundle
+
+To enable it, add the following line to the ``config/bundles.php`` file,
+if that has not been done already for you by Symfony Flex:
+
+.. code:: php
+
+   return [
+       ...
+       Symfony\WebpackEncoreBundle\WebpackEncoreBundle::class => ['all' => true],
+   ];
+
+Configuration
+-------------
+
+There are several files and directories added to your project by Symfony
+Flex:
+
+* ``assets/css/``
+* ``assets/js/``
+* ``config/packages/assets.yaml``
+* ``config/packages/prod/webpack_encore.yaml``
+* ``config/packages/test/webpack_encore.yaml``
+* ``config/packages/webpack_encore.yaml``
+* ``package.json``
+* ``webpack.config.js``
+
+In a normal Symfony application, that would already work fine, but to
+use Webpack Encore with Sulu, some configuration has to be adjusted,
+because there is an additional build for Sulu’s admin interface.
+
+To continue, create a directory ``assets/website/`` and move the
+following directories and files into that directory:
+
+* ``assets/css/``
+* ``assets/js/``
+* ``package.json``
+* ``webpack.config.js``
```

Comment:
I would not move this files as symfony recipes update for this files could be important and so I would keep them in the root folder of the project.

Reply:
I would rather add another section **Upgrading the Webpack Encore recipe**. Upgrading will never be straightforward, because flex will override all our changed files during the upgrade (all configuration files, app.js, app.css, package.json ...), so you would have to configure all of them again. @alexander-schranz @nnatter what do you think?

Reply:
I would stay the symfony way as near as possible, don't see any advantages to force them that they need to go into the assets/website folder to run npm scripts for their application.

Reply:
I'm not really happy with that, because I can't see a benefit in that change, but I implemented it as you wanted.

Reply:
Its consistent for the developers which use symfony and sulu projects having the npm scripts for there application at the same places.